Description: include explicitly IT instructions to fix build failure without
implicit-it=thumb
Author: Michael Hope <michael.hope@linaro.org>
Bug-Ubuntu: https://launchpad.net/bugs/673085
---
 src/corelib/arch/qatomic_armv6.h |    2 ++
 1 file changed, 2 insertions(+)

--- a/src/corelib/arch/qatomic_armv6.h
+++ b/src/corelib/arch/qatomic_armv6.h
@@ -144,6 +144,7 @@ inline bool QBasicAtomicInt::testAndSetO
     asm volatile("0:\n"
                  "ldrex %[result], [%[_q_value]]\n"
                  "eors %[result], %[result], %[expectedValue]\n"
+                 "itt eq\n"
                  "strexeq %[result], %[newValue], [%[_q_value]]\n"
                  "teqeq %[result], #1\n"
                  "beq 0b\n"
@@ -202,6 +203,7 @@ Q_INLINE_TEMPLATE bool QBasicAtomicPoint
     asm volatile("0:\n"
                  "ldrex %[result], [%[_q_value]]\n"
                  "eors %[result], %[result], %[expectedValue]\n"
+                 "itt eq\n"
                  "strexeq %[result], %[newValue], [%[_q_value]]\n"
                  "teqeq %[result], #1\n"
                  "beq 0b\n"
